src/WebJobs.Script.WebHost/WebHostServiceCollectionExtensions.cs

```diff
@@ -188,6 +187,52 @@ private static void AddLinuxContainerServices(this IServiceCollection services)
                 var nullMetricsLogger = s.GetService<ILogger<NullMetricsPublisher>>();
                 return new NullMetricsPublisher(nullMetricsLogger);
             });
+
+            services.AddSingleton<IMeshInitServiceClient>(s =>
+            {
+                var environment = s.GetService<IEnvironment>();
+                if (environment.IsLinuxConsumption())
+                {
+                    var httpClient = s.GetService<HttpClient>();
+                    var logger = s.GetService<ILogger<MeshInitServiceClient>>();
+                    return new MeshInitServiceClient(httpClient, environment, logger);
```

Comment:
I suspect that all these places where we're newing the service up and passing in resolved instances, we could just allow DI to do that for us, e.g. GetRequiredService<MeshInitServiceClient>()?

Reply:
@Fabio - likely there is a cleaner pattern that we can use for this. Because we're doing conditional adds for all these Linux services, we're forced to new the service instances up ourselves, passing in ctor params (anti-pattern). Is there a better way?